### PR TITLE
HELPS-3348: Add NPX starter and VS Code Extension tracking in GA

### PR DESCRIPTION
**Pull request by JosefBud-olive**
### Changes made:

- Changed `create-loop` from a pure JS Node script to a compiled TS Node script
  - Now that both create-loop and the extension use a build process, `loop-templates` is now relational only (no longer pulled from NPM) and compiled into each one during the CI build process
  - Changed README to reflect this new way of using `loop-templates`
- Added Google analytics tracking in NPX and VSCode extension
- Added environment and analytics ID injection to CI build process
- Added ESLint to `create-loop` package

### CODE REVIEW

File: create-loop/src/index.ts [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
+  ua(__GOOGLE_ANALYTICS_ID__) : 
+  // For development
+  {
+    event: () => ({
+        send: () => undefined
+      }),
+    set: () => undefined,
+  };
+
+const osName = platform === 'win32' ? 'windows' : platform;
```
Comment (MaggieMaSS) [inline create-loop/src/index.ts:24]: May I know why we specify win32 here? 🤔

Comment (MaggieMaSS) [reply create-loop/src/index.ts:24]: like why we set cd1 for all other environments and cd4 only for windows?

Comment (JosefBud-olive) [reply create-loop/src/index.ts:24]: `cd1` is for the environment, e.g. "dev" vs "prod"

`cd4` is for the OS name. For some reason we specify Windows as `win32` in the rest of our analytics for Helps, so I'm just following what they did :)

Comment (JosefBud-olive) [reply create-loop/src/index.ts:24]: https://gitlab.oliveai.com/oliveai/platform/desktop/ui/-/blob/main/src/main/analytics/index.ts#L69 Here's where UI does the same if you're interested! :)

Comment (MaggieMaSS) [reply create-loop/src/index.ts:24]: Ahhh thx for the explanation!

**Review by MaggieMaSS — APPROVED** [review]

**Review by BrettMugglinOlive — APPROVED** [review]

**Review by iangallOlive — APPROVED** [review]

**Pull request merged**